app/services/forest_liana/scope_validator.rb

```diff
@@ -0,0 +1,94 @@
+module ForestLiana
+  class ScopeValidator
+    def initialize(scope_permissions, users_variable_values)
+      @scope_filters = scope_permissions
+      @users_variable_values = users_variable_values
+    end
+
+    def is_scope_in_request?(scope_request)
+      begin
+        filters = JSON.parse(scope_request[:filters])
+      rescue JSON::ParserError
+        raise ForestLiana::Errors::HTTP422Error.new('Invalid filters JSON format')
+      end
+      @computed_scope = compute_condition_filters_from_scope(scope_request[:user_id])
+
+      # NOTICE: Perfom a travel in the request condition filters tree to find the scope
+      tagged_scope_filters = validate(filters)
+
+      # NOTICE: Permission system always send an aggregator even if there is only one condition
+      #         In that case, if the condition is valid, then request was not edited
+      return tagged_scope_filters != nil if @scope_filters['conditions'].length == 1
+
+      # NOTICE: If there is more than one condition, do a final validation on the condition filters
+      return tagged_scope_filters != nil &&
+        tagged_scope_filters[:aggregator] == @scope_filters['aggregator'] &&
+        tagged_scope_filters[:conditions] &&
+        tagged_scope_filters[:conditions].length == @scope_filters['conditions'].length
+    end
+
+    private
+
+    def compute_condition_filters_from_scope(user_id)
+      computed_condition_filters = @scope_filters.clone
+      computed_condition_filters['conditions'].each do |condition|
+        if condition.include?('value') && condition['value'].start_with?('$') && @users_variable_values.include?(user_id)
+          condition['value'] = @users_variable_values[user_id][condition['value']]
+        end
+      end
+      return computed_condition_filters
+    end
+
+    def validate(filters)
+      return nil unless filters
+      return search_scope_aggregation(filters)
+    end
+
+    def search_scope_aggregation(node)
+      ensure_valid_aggregation(node)
+
+      return is_scope_condition?(node) unless node['aggregator']
+  
+      # NOTICE: Remove conditions that are not from the scope
+      filtered_conditions = node['conditions'].map { |condition| 
+        search_scope_aggregation(condition)
+      }.select { |condition|
+        condition
+      }
+
+      # NOTICE: If there is only one condition filter left and it's current aggregator is
+      #         an "and", this condition filter is the searched scope
+      if (filtered_conditions.length === 1 && 
```

Comment:
[MISC] As stated [here](https://stackoverflow.com/questions/3422223/vs-in-ruby), you should avoid the `===` comparison operator as it does not have the same meaning as in JS

```suggestion
      if (filtered_conditions.length == 1 && 
```